upf/conditional_effects_remover.py

```diff
@@ -0,0 +1,171 @@
+        returns False, otherwise returns True.
+        If the simplification is True (a tautology) removes all preconditions.
+        If the simplification is still an AND rewrites back every "arg" of the AND
+        in the preconditions
+        If the simplification is not an AND sets the simplification as the only
+        precondition.'''
+        action_preconditions = action.preconditions()
+        if len(action_preconditions) == 0:
+            return True
+        precondition = self._env.expression_manager.And(action_preconditions)
+        precondition_simplified = self._simplifier.simplify(precondition)
+        if precondition_simplified.is_bool_constant():
+            if precondition_simplified.bool_constant_value():
+                action_preconditions.clear()
+            else:
+                return False
+        else:
+            action_preconditions.clear()
+            if precondition_simplified.is_and():
+                action_preconditions.extend(precondition_simplified.args())
+            else:
+                action_preconditions.append(precondition_simplified)
+        return True
+
+    def _create_action_copy(self, action, original_action_name, unchanged_effects, uncond_action) -> Action:
+        if uncond_action: #search for a free name
+            #emulates a do-while loop
+            is_unavailable_name = True
+            while is_unavailable_name:
+                new_action_name = original_action_name+ "_" +str(self._counter)
+                self._counter = self._counter + 1
+                is_unavailable_name = self._problem.has_action(new_action_name)
+        else: #otherwise the name does not matter.
+            new_action_name = original_action_name
+
+        new_parameters = OrderedDict()
+        for ap in action.parameters():
+            new_parameters[ap.name()] = ap.type()
+        new_action = Action(new_action_name, new_parameters, self._env)
+        for p in action.preconditions():
+            new_action.add_precondition(p)
+        for e in unchanged_effects:
+            new_action._add_effect_instance(e)
+        return new_action
+
+    def _create_problem_copy(self, action_stack) -> Problem:
+        '''Creates the shallow copy of a problem, without adding the conditional actions
+        and by pushing them to the stack
+        '''
+        new_problem: Problem = Problem("unconditional_" + str(self._problem.name()), self._env)
+        for f in self._problem.fluents().values():
+            new_problem.add_fluent(f)
+        for o in self._problem.all_objects():
+            new_problem.add_object(o)
+        for fl, v in self._problem.initial_values().items():
+            new_problem.set_initial_value(fl, v)
+        for g in self._problem.goals():
+            new_problem.add_goal(g)
+        for n, a in self._problem.actions().items():
+            if a.has_conditional_effects():
+                action_stack.append((n, a))
```

Comment:
In python, it's more common to return a list rather than filling a list taken in input